synapseclient/models/submission.py

```diff
@@ -0,0 +1,939 @@
+                async for submission in Submission.get_evaluation_submissions_async(
+                    evaluation_id="9999999",
+                    status="SCORED"
+                ):
+                    submissions.append(submission)
+                print(f"Found {len(submissions)} submissions")
+
+            asyncio.run(get_evaluation_submissions_example())
+            ```
+        """
+        async for submission_data in evaluation_services.get_evaluation_submissions(
+            evaluation_id=evaluation_id,
+            status=status,
+            synapse_client=synapse_client,
+        ):
+            submission_object = cls().fill_from_dict(synapse_submission=submission_data)
+            yield submission_object
+
+    @skip_async_to_sync
+    @classmethod
+    async def get_user_submissions_async(
+        cls,
+        evaluation_id: str,
+        user_id: Optional[str] = None,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> AsyncGenerator["Submission", None]:
+        """
+        Generator to get all user Submissions for a specified Evaluation queue.
+        If user_id is omitted, this returns the submissions of the caller.
+
+        Arguments:
+            evaluation_id: The ID of the evaluation queue.
+            user_id: Optionally specify the ID of the user whose submissions will be returned.
+                    If omitted, this returns the submissions of the caller.
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Yields:
+            Individual Submission objects from each page of the response.
+
+        Example: Getting user submissions
+            ```python
+            import asyncio
+            from synapseclient import Synapse
+            from synapseclient.models import Submission
+
+            syn = Synapse()
+            syn.login()
+
+            async def get_user_submissions_example():
+                submissions = []
+                async for submission in Submission.get_user_submissions_async(
+                    evaluation_id="9999999",
+                    user_id="123456"
+                ):
+                    submissions.append(submission)
+                print(f"Found {len(submissions)} user submissions")
+
+            asyncio.run(get_user_submissions_example())
+            ```
+        """
+        async for submission_data in evaluation_services.get_user_submissions(
+            evaluation_id=evaluation_id,
+            user_id=user_id,
+            synapse_client=synapse_client,
+        ):
+            submission_object = cls().fill_from_dict(synapse_submission=submission_data)
+            yield submission_object
+
+    @staticmethod
+    async def get_submission_count_async(
+        evaluation_id: str,
+        status: Optional[str] = None,
```

Comment:
Same comment as above: I think we should add validation to ensure that `status` is one of the allowed enum values.